frame/bridge/s2s/chain/src/lib.rs

```diff
@@ -0,0 +1,82 @@
+// This file is part of Darwinia.
+//
+// Copyright (C) 2018-2021 Darwinia Network
+// SPDX-License-Identifier: GPL-3.0
+//
+// Darwinia is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Darwinia is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Darwinia. If not, see <https://www.gnu.org/licenses/>.
+
+//! s2s chain info include runtime and call method.
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use darwinia_asset_primitives::token::Token;
+use darwinia_relay_primitives::RelayAccount;
+use ethereum_primitives::EthereumAddress;
+use sp_std::vec::Vec;
+
+// here we must contruct a Backing Runtime Call to call backing pallet from the remote issuing
+// pallet, because also we have the other direction call from backing pallet to this issuing
+// pallet. Otherwise if we import the backing pallet and use the Call definiation. The must be a circular reference
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+pub enum MillauRuntime {
+	/// s2s bridge backing pallet.
+	#[codec(index = 49)]
+	Sub2SubBacking(MillauSub2SubBackingCall),
+}
+//
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+#[allow(non_camel_case_types)]
+pub enum MillauSub2SubBackingCall {
+	#[codec(index = 0)]
+	cross_receive((Token, EthereumAddress)),
+}
+
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+pub enum PangolinRuntime {
+	Sub2SubIssuing(PangolinSub2SubIssuingCall),
+}
+
+#[derive(Encode, Decode, Debug, PartialEq, Eq, Clone)]
+#[allow(non_camel_case_types)]
+pub enum PangolinSub2SubIssuingCall {
+	#[codec(index = 0)]
+	cross_receive((Token, EthereumAddress)),
+}
+
+pub type ChainSelector = [u8; 4];
+
+pub fn encode_relay_message<AccountId>(
+	selector: ChainSelector,
```

Comment:
Do we really have to use chain selector here?

Reply:
yes, we use it to tell us which chain to receive message. and this will be moved to runtime.

Reply:
Ah, I see, but fro each mapping token, we know its source chain right?

Reply:
yes. the target chain id is unique.